tests/testthat/test-shinytest2-tm_g_scatterplot.R

```diff
@@ -0,0 +1,194 @@
+  data <- teal.data::teal_data()
+  data <- within(data, {
+    require(nestcolor)
+    ADSL <- rADSL
+  })
+  teal.data::datanames(data) <- c("ADSL")
+  teal.data::join_keys(data) <- teal.data::default_cdisc_join_keys[teal.data::datanames(data)]
+
+  init_teal_app_driver(
+    data = data,
+    modules = teal::modules(
+      tm_g_scatterplot(
+        label = "Scatterplot Choices",
+        x = teal.transform::data_extract_spec(
+          dataname = "ADSL",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(data[["ADSL"]], c("AGE", "BMRKR1", "BMRKR2")),
+            selected = "AGE",
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        y = teal.transform::data_extract_spec(
+          dataname = "ADSL",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(data[["ADSL"]], c("AGE", "BMRKR1", "BMRKR2")),
+            selected = "BMRKR1",
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        color_by = data_extract_spec(
+          dataname = "ADSL",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(
+              data[["ADSL"]],
+              c("AGE", "BMRKR1", "BMRKR2", "RACE", "REGION1")
+            ),
+            selected = NULL,
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        size_by = data_extract_spec(
+          dataname = "ADSL",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(data[["ADSL"]], c("AGE", "BMRKR1")),
+            selected = "AGE",
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        row_facet = data_extract_spec(
+          dataname = "ADSL",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(data[["ADSL"]], c("BMRKR2", "RACE", "REGION1")),
+            selected = NULL,
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        col_facet = data_extract_spec(
+          dataname = "ADSL",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(data[["ADSL"]], c("BMRKR2", "RACE", "REGION1")),
+            selected = NULL,
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        ggplot2_args = teal.widgets::ggplot2_args(
+          labs = list(subtitle = "Plot generated by Scatterplot Module")
+        )
+      )
+    )
+  )
+}
+
+testthat::test_that("e2e - tm_g_scatterplot: module is initialised with the specified defaults", {
```

Comment:
Maybe we can only test the options here that are configurable by the module?

For instance, `add_density`, `rug_plot`, `show_count` are hard-coded.

Reply:
Applied in 4d8cb4e0